tests/test_exposure.py

```diff
@@ -777,6 +777,88 @@ def testGetCutout(self):
                     with self.assertRaises(pexExcept.InvalidParameterError, msg=msg):
                         self.smallExposure.getCutout(cutoutCenter, cutoutSize)
 
+    def testGetConvexPolygon(self):
+        """Test the convex polygon."""
+        # Check that we do not have a convex polygon for the plain exposure.
+        self.assertIsNone(self.exposureMiOnly.convex_polygon)
+
+        # Check that all the points in the bounding box are in the polygon
+        bbox = self.exposureMiWcs.getBBox()
+        # Grow by the default padding.
+        bbox.grow(10)
+        x, y = np.meshgrid(np.arange(bbox.getBeginX(), bbox.getEndX()),
+                           np.arange(bbox.getBeginY(), bbox.getEndY()))
+        wcs = self.exposureMiWcs.wcs
+        ra, dec = wcs.pixelToSkyArray(x.ravel().astype(np.float64),
+                                      y.ravel().astype(np.float64))
+
+        poly = self.exposureMiWcs.convex_polygon
+        contains = poly.contains(ra, dec)
+        np.testing.assert_array_equal(contains, np.ones(len(contains), dtype=bool))
+
+        # Check that points outside of the bounding box are not in the polygon
+        bbox.grow(1)
+
+        ra, dec = wcs.pixelToSkyArray(
```

Comment:
I'm a bit confused by all of these different versions. I think this is testing each side of the box? Maybe making a small closure would make the differences clearer? It might not help, though.

```
def check_outside(x, y):
    """Check that the x,y points are all outside the polygon."""
    ra, dec = wcs.pixelToSkyArray(x, y)
    contains = poly.contains(ra, dec)
    np.testing.assert_array_equal(contains, np.zeros(len(contains), dtype=bool))
```

Reply:
So I spent about 2 minutes trying to come up with a cleverer version of getting the perimeter and decided that it would be more trouble than its worth for a simple test.